provider/cloud.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 Giant Swarm.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"io"
+	"os"
+
+	"github.com/golang/glog"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+const providerName string = "kubernetes"
+const providerKubeconfigEnv string = "PROVIDER_HOST_CLUSTER_KUBECONFIG"
+const providerNamespaceEnv string = "PROVIDER_HOST_CLUSTER_NAMESPACE"
+
+type cloud struct {
+	client    clientset.Interface
+	instances cloudprovider.Instances
+}
+
+func newCloud(config io.Reader) (cloudprovider.Interface, error) {
```

Comment:
Why this takes `config` argument?

Reply:
Please see comment above. But, yes, it's not used here.